### PR TITLE
GRAPHICS: Move common OpenGL files to graphics/opengl

### PR DESCRIPTION
**Pull request by Botje**
This PR moves common OpenGL classes to graphics/opengl and adds a `Graphics::Texture`class which embodies an OpenGL texture.
This is part cleanup, and part preparation for #862 .

### CODE REVIEW

**Comment by aquadran** [comment]
:+1:

File: graphics/opengl/texture.cpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+Texture::Texture(const Surface &srf) :
+		_managedTexture(true), _width(srf.w), _height(srf.h),
+		_texWidth(nextHigher2(_width)), _texHeight(nextHigher2(_height)) {
+	glGenTextures(1, &_texture);
+	glBindTexture(GL_TEXTURE_2D, _texture);
+	glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_REPEAT);
+	glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_REPEAT);
+	glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
+	glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+	// FIXME: what if buffer is not RGBA?
```
Comment (bgK) [inline graphics/opengl/texture.cpp]: Maybe assert the surface has the expected pixel format for now?

**Comment by bgK** [comment]
:+1:

File: graphics/opengl/texture.cpp [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA, _texWidth, _texHeight, 0, GL_RGBA, GL_UNSIGNED_BYTE, 0);
+	glTexSubImage2D(GL_TEXTURE_2D, 0, 0, 0, _width, _height, GL_RGBA, GL_UNSIGNED_BYTE, srf.getPixels());
```
Comment (bgK) [inline graphics/opengl/texture.cpp]: Will that work on big endian systems?

**Comment by Botje** [comment]
Addressed the comments. Good to go?

**Pull request merged**